### PR TITLE
Fixes #10236 - Fix wrong error message when creating second primary/provision interface

### PR DESCRIPTION
**Pull request by yifatmakias**

### CODE REVIEW

File: app/models/nic/base.rb [hunk 1]
```diff
@@ -318,14 +318,14 @@ def not_required_interface
     def exclusive_primary_interface
       if host && primary?
         primaries = host.interfaces.select { |i| i.primary? && i != self }
-        errors.add :primary, _("host already has primary interface") unless primaries.empty?
+        errors.add :primary, _("interface is already set to the host") unless primaries.empty?
```
Comment (tbrisker) [inline app/models/nic/base.rb]: How does it show up in the UI if you create two of a kind? And a minor grammar correction:
```suggestion
        errors.add :primary, _("interface is already set on the host") unless primaries.empty?
```

File: app/models/nic/base.rb [hunk 2]
```diff
@@ -318,14 +318,14 @@ def not_required_interface
     def exclusive_primary_interface
       if host && primary?
         primaries = host.interfaces.select { |i| i.primary? && i != self }
```
Comment (tbrisker) [inline app/models/nic/base.rb]: just a thought while we're here, we don't necessarily need to iterate over all and create a new array, we could do something like:
```suggestion
        duplicate = host.interfaces.any? { |i| i.primary? && i != self }
```
and then `error... if duplicate` (and same for provision)

File: app/models/nic/base.rb (hunk 1 above)
Comment (yifatmakias) [reply app/models/nic/base.rb]: In the UI it looks like this:
image

**Review by tbrisker — COMMENTED** [review]
Thanks @yifatmakias ! one last question: what does the error message look like in hammer after this change?

**Comment by yifatmakias** [comment]
> Thanks @yifatmakias ! one last question: what does the error message look like in hammer after this change?

image

**Review by tbrisker — APPROVED** [review]

**Pull request merged**